python-backend/app/api/mines/compliance/resources/compliance.py

```diff
@@ -8,65 +9,74 @@
-                stops_dict = stops[0]
-
-                stop_orders = stops_dict.get('stopOrders')
-                stop_advisories = stops_dict.get('stopAdvisories')
-                stop_warnings = stops_dict.get('stopWarnings')
-
-                if stop_orders:
-                    open_orders += sum(
-                        k.get('orderStatus') == 'Open' for k in stop_orders)
-                    overdue_orders += sum(
-                        k.get('orderStatus') == 'Open' and datetime.strptime(
-                            k.get('orderCompletionDate'), '%Y-%m-%d %H:%M') <
-                        datetime.now() for k in stop_orders)
-                    section_35_orders += sum(
-                        k.get('orderAuthoritySection') == 'Section 35'
-                        for k in stop_orders)
-
-                if one_year_ago < report_date:
-                    advisories += len(stop_advisories)
-                    warnings += len(stop_warnings)
-
-        overview = {
-            'last_inspection': most_recent.get('assessmentDate'),
-            'inspector': most_recent.get('assessor'),
-            'open_orders': open_orders,
-            'overdue_orders': overdue_orders,
-            'advisories': advisories,
-            'warnings': warnings,
-            'section_35_orders': section_35_orders,
-        }
-
-        return overview
+        try:
+            data = NRIS_service.get_EMPR_data_from_NRIS(mine_no)
+        except requests.exceptions.HTTPError as errhttp:
+            return self.raise_error(
+                errhttp.response.status_code,
+                "There has been an unexpected error with NRIS.")
+        except requests.exceptions.Timeout as errt:
+            return self.raise_error(408, "NRIS has timed out.")
+
+        if len(data) == 0:
+            return None
+        else:
+            data = sorted(
+                data, key=lambda k: datetime.strptime(k.get('assessmentDate'), '%Y-%m-%d %H:%M'), reverse=True)
+
+            most_recent = data[0]
+
+            advisories = 0
+            warnings = 0
+            open_orders = 0
+            overdue_orders = 0
+            section_35_orders = 0
+
+            for report in data:
+                report_date = datetime.strptime(
+                    report.get('assessmentDate'), '%Y-%m-%d %H:%M')
+                one_year_ago = datetime.now() - relativedelta(years=1)
+
+                inspection = report.get('inspection')
+                stops = inspection.get('stops')
+
+                if stops:
+                    stops_dict = stops[0]
+
+                    stop_orders = stops_dict.get('stopOrders')
+                    stop_advisories = stops_dict.get('stopAdvisories')
+                    stop_warnings = stops_dict.get('stopWarnings')
+
+                    if stop_orders:
+                        open_orders += sum(
+                            k.get('orderStatus') == 'Open'
+                            for k in stop_orders)
```

Comment:
A `lambda` seems more appropriate here. It would also avoid the type coercion. Something like...
```python
open_orders = sum(map(
  lambda k: 1 if k.get('orderStatus') == 'Open' else 0,
  stop_orders
))
```

Reply:
I don't see how the lambda would be better. List comprehension is usually the faster way when using a non defined function or logical expression to evaluate each element. Im also not sure about the type coercion as the list this is being run on is part of a JSON return and the field itself in the NRIS definitions is a string.

Reply:
My point about the type coercion was just that you're implicitly coercing booleans to zeros and ones. It's not a bit deal, but it does obfuscate the purpose of this code block a bit in my opinion

I'm not familiar with how python structures list comprehensions. This doesn't looks like list comprehensions that I'm familiar with. This looks very procedural, particularly with the `+=` operator.

I'm not going to push back super hard on this point, as I'm not familiar with what python considers to be most idiomatic.